xtask/src/main.rs

```diff
@@ -0,0 +1,284 @@
+    }
+}
+
+impl Ord for EfuseAttrs {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.block.cmp(&other.block) {
+            Ordering::Equal => {}
+            ord => return ord,
+        }
+
+        match self.word.cmp(&other.word) {
+            Ordering::Equal => {}
+            ord => return ord,
+        }
+
+        self.start.cmp(&other.start)
+    }
+}
+
+fn generate_efuse_fields(workspace: &Path, args: GenerateEfuseFieldsArgs) -> Result<()> {
+    let efuse_yaml_path = args
+        .esptool_path
+        .join("espefuse")
+        .join("efuse_defs")
+        .canonicalize()?;
+
+    let espflash_path = workspace.join("espflash").canonicalize()?;
+
+    let mut efuse_fields = parse_efuse_fields(&efuse_yaml_path)?;
+    process_efuse_definitions(&mut efuse_fields)?;
+    generate_efuse_definitions(&espflash_path, efuse_fields)?;
+
+    Command::new("cargo")
+        .args(["+nightly", "fmt"])
+        .current_dir(workspace)
+        .output()?;
+
+    Ok(())
+}
+
+fn parse_efuse_fields(efuse_yaml_path: &Path) -> Result<EfuseFields> {
+    // TODO: We can probably handle this better, e.g. by defining a `Chip` enum
+    //       which can be iterated over, but for now this is good enough.
+    const CHIPS: &[&str] = &[
+        "esp32", "esp32c2", "esp32c3", "esp32c6", "esp32h2", "esp32p4", "esp32s2", "esp32s3",
+    ];
+
+    let mut efuse_fields = EfuseFields::new();
+
+    for result in fs::read_dir(efuse_yaml_path)? {
+        let path = result?.path();
+        if path.extension().is_none_or(|ext| ext != OsStr::new("yaml")) {
+            continue;
+        }
+
+        let chip = path.file_stem().unwrap().to_string_lossy().to_string();
+        if !CHIPS.contains(&chip.as_str()) {
+            continue;
+        }
+
+        let efuse_yaml = fs::read_to_string(&path)?;
+        let efuse_yaml: EfuseYaml = serde_yaml::from_str(&efuse_yaml)?;
+
+        efuse_fields.insert(chip.to_string(), efuse_yaml);
+    }
+
+    Ok(efuse_fields)
+}
+
+fn process_efuse_definitions(efuse_fields: &mut EfuseFields) -> Result<()> {
+    // This is all a special case for the MAC field, which is larger than a single
+    // word (i.e. 32-bits) in size. To handle this, we just split it up into two
+    // separate fields, and update the fields' attributes accordingly.
+    for yaml in (*efuse_fields).values_mut() {
+        let mac_attrs = yaml.fields.get("MAC").unwrap();
+
+        let mut mac0_attrs = mac_attrs.clone();
+        mac0_attrs.start = 0;
+        mac0_attrs.len = 32;
+
+        let mut mac1_attrs = mac_attrs.clone();
+        mac1_attrs.word += 1;
+        mac1_attrs.len = 16;
+
+        yaml.fields.remove("MAC").unwrap();
+        yaml.fields.insert("MAC0".into(), mac0_attrs);
+        yaml.fields.insert("MAC1".into(), mac1_attrs);
+    }
+
+    // The ESP32-S2 seems to be missing a reserved byte at the end of BLOCK0
+    // (Or, something else weird is going on).
+    for (chip, yaml) in efuse_fields {
```

Comment:
I don't think this needs to be a loop, I think you can just grab the esp32s2 fields from the `EfuseFields` hashmap and modify it once?

Reply:
Yeah was just copy-pasting and being lazy